core/inc/knoting/light_data.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <bgfx/bgfx.h>
+#include <knoting/types.h>
+
+namespace knot {
+class LightData {
+   private:
+    class SpotlightData {
```

Comment:
this is mainly to have the various 'xxLightData' classes hidden so all you need to use is the API, but I am happy to do something else.
could have lots of 'xxLightData' classes instead of them being nested inside of 'LightData' would just mean there are more classes in the ForwardRenderer, but it would certainly be very clear what data is used where c:

Reply:
I find nested classes a little hard to read. Having a `SpotLightData`, or other light data classes outside shouldn't pollute the namespace that much.